### ISSUE TITLE
ZPM installation returns "SQL error(s) have occurred during the running of the 'statsINFO()' method"

### ISSUE THREAD

**Issue by pisani** [intersystems-community/zpm#334]
Trying to install a ZPM package on IRIS 2021.2 build 649. Using latest ZPM generates the error:

```
ERROR! ObjectScript error: SQL error(s) have occurred during the running of the 'statsINFO()' method.  Invoke the following method for additional details: set errID=##class(%SYS.PTools.Stats).getPToolsError(...)  WHERE errRowID={57|58}>
> ERROR #6084: Unknown errors detected, but no error code reported
```

The module I am using is "git-source-control", but I also tried "obj-dump" and got the same error. I do not think this is module related.  This happens in all namespaces.

This was also raised in this community post - https://community.intersystems.com/post/errors-installing-openexchange-git-source-control-using-zpm, and as mentioned there with no resolution.  It has also been reported as issue 137 under the module 'git-source-control's repository and I've attempted some diagnositics by looking there. 

I've attempting to unpack the error SQL Stats error:  

```
zpm: install obj-dump
ERROR! ObjectScript error: SQL error(s) have occurred during the running of the 'statsINFO()' method.  Invoke the following method for additional details: set errID=##class(%SYS.PTools.Stats).getPToolsError(...)  WHERE errRowID={89|90}>
> ERROR #6084: Unknown errors detected, but no error code reported
zpm:DICS>exit
 
DICS>do ##class(%SYS.PTools.Stats).getPToolsError(89,.errArray1)
 
DICS>zw errArray1
errArray1(89,0)=$lb("66324,42052","2022-08-03 11:40:52","DICS","52244","%SYS.PTools",65)
errArray1(89,1)=$lb("DICS","52244","%SYS.sqlcq.xE1tuuQfilDLKNL6tWR40kxdMHI4.1","A","","statsINFO","","SQL Error: Failed to SELECT from <table>INFORMATION_SCHEMA.STATEMENTS</table>. The data from this table is requied to re-create/INSERT the 'INFO' row (ModuleName = 'INFO') for statistics collection.","INFORMATION.SCHEMA.STATEMENTS","INFORMATION_SCHEMA.STATEMENTS",100,"No (more) data")
errArray1(89,2,"SQLIndexHash")="ti9E74nMywiuJGdaDzSj37wXp8M="
errArray1(89,2,"rtnInfo")="DICS|%SYS.sqlcq.xE1tuuQfilDLKNL6tWR40kxdMHI4.1|A|0"
> ERROR! ObjectScript error: SQL error(s) have occurred during the running of the 'statsINFO()' method.  Invoke the following method for additional details: set errID=##class(%SYS.PTools.Stats).getPToolsError(...)  WHERE errRowID={89|90}>
 ERROR #6084: Unknown errors detected, but no error code reported
zpm:DICS>exit
```
 
The table INFORMATION_SCHEMA.STATEMENTS exists, and I can SELECT columns from it.  If - however, the error is saying ZPM is trying to insert/retrieve a row in this table with Column named 'ModuleName' and value 'INFO' - then, I do not see this column in that table.   I'm not sure that is the issue though.

**Comment by daimor** [comment]
Hi Steve, I have no idea how to reproduce this, yet. 
On my fresh instance, works perfectly for both packages.
Could you somehow, find a way, to get the STACKTRACE, so, I could try to find, the last place in ZPM. At this moment, I can't find anything similar in ZPM's code, so, it may happen in somewhere outside.

**Comment by daimor** [comment]
Could you add the $ZV, please?

**Comment by pisani** [comment]
ok.. this is strange !

Firstly - $zv = IRIS for Windows (x86-64) 2021.2 (Build 649U) Thu Jan 20 2022 08:46:32 EST - I think you are going to need this....

Now I trying to find STACKTRACE, I started new process, entered into ZPM shell, and generated error - which failed exactly as reported above. - so I ran JOBEXAM from another process to get trace but does not show much:

`Retrieving stack information for process 73212
Process Stack:

Level  Type     Line                   Source
  1   SIGN ON                            
  2   DO                                 ~ZPM
  3   PARAMETER                          pArgs
  4   DO                    ZPM^%ZLANGC00 ZPM(pArgs...) Do ##class(%ZPM.PackageManager).Shell(pArgs...) ~Quit
  5   PARAMETER                          ?,?,?
  6   DO               zShell+2^%ZPM.PackageManager.1          ~Do ..ShellInternal(.pCommand,.tException)
  7   PARAMETER                          .?,.?
  8   FOR          zShellInternal+12^%ZPM.PackageManager.1          ~For {
  9   XECUTE       zShellInternal+22^%ZPM.PackageManager.1     ~Read tCommand
`
But then - I decided to dive into ZPM to see if I can help - I after several attempts have tracked it down to see it is being raised by $SYSTEM.OBJ.Load here:

class: %ZPM.PackageManager.Developer.Utils 
method: LoadNewModule
line: (940)

`Set tSC = $system.OBJ.Load(pDirectory_"module.xml",$Select(tVerbose:"d",1:"-d"),,.tLoadedList)`

Which is very strange...  but - I ran the same line of code, in terminal and I get the same error reported.
pDirectory = "C:\InterSystems\IRIS\mgr\.modules\DICS\obj-dump\1.0.8\"
tVerbose = ""

Of course - this is very strange.  I cannot currently see the contents of $$Load^%apiOBJ(...) but this is where we are..  These seems to be an issue with the system code on this particular version !
